packages/itwin/grouping-mapping-widget/src/test/QueryBuilder.test.ts

```diff
@@ -2,178 +2,67 @@
-    });
-
-    it("should return query string with where clause, when base class has properties", () => {
-      const baseClass: QueryClass = {
-        className: "BaseClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [
-          {
-            name: "propName1",
-            needsQuote: false,
-            isCategory: false,
-            value: "propValue1",
-          },
-          {
-            name: "propName2",
-            needsQuote: false,
-            isCategory: false,
-            value: "propValue2",
-          },
-        ],
-      };
-
-      const joinClass: QueryClass = {
-        className: "JoinClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [],
-      };
-
-      const query: Query = {
-        unions: [
-          {classes: [baseClass, joinClass]},
-        ],
-      };
-      testBuildQueryString(query, "SELECT BaseClassName.ECInstanceId FROM BaseClassName JOIN JoinClassName WHERE BaseClassName.propName1=propValue1 AND BaseClassName.propName2=propValue2");
-    });
-
-    it("should return query string with UNION, when there are multiple unions", () => {
-      const queryClass: QueryClass = {
-        className: "ClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [
-          {
-            name: "propName1",
-            needsQuote: false,
-            isCategory: false,
-            value: "propValue1",
-          },
-        ],
-      };
+    sut = new QueryBuilder(dataProvider);
+  });
 
-      const query: Query = {
-        unions: [
-          {classes: [queryClass]},
-          {classes: [queryClass]},
-        ],
-      };
+  afterEach(() => {
+    dataProvider.getFieldByPropertyRecord.restore();
+  });
 
-      testBuildQueryString(query, "SELECT ClassName.ECInstanceId FROM ClassName WHERE ClassName.propName1=propValue1 UNION SELECT ClassName.ECInstanceId FROM ClassName WHERE ClassName.propName1=propValue1");
-    });
+  const testData: QueryBuilderTestData = testCases;
+  testData.testCases.forEach((testCase) => {
+    it(testCase.name, async () => executeTest(
+      sut,
+      testCase.expectedResult,
+      testCase.operations,
+    ));
   });
+
+  const createPropertyRecord = (propertyRecord: PropertyRecordTestData, propertiesField: PropertiesField) => {
+    const propertiesFieldMock: StubbedType<PropertiesField> = MockFactory.create(PropertiesField);
+
+    MockFactory.stubProperty(propertiesFieldMock, "parent", () => propertiesField.parent);
+    MockFactory.stubProperty(propertiesFieldMock, "properties", () => propertiesField.properties);
+    MockFactory.stubProperty(propertiesFieldMock, "type", () => propertiesField.type);
+
+    const prop: PropertyRecord = new PropertyRecord(propertyRecord.value as PropertyValue, propertyRecord.property as PropertyDescription);
```

Comment:
Are these casts required? Maybe some of the `TestData` interfaces are not required and types from Presentation libs could be reused.

Reply:
Why did you leave `op.propertiesField as PropertiesField` unchanged?

Reply:
`PropertiesField` is a huge object with lots of internals, changing from `PropertiesFieldTestData` to `PropertiesField` would bloat test data file a lot, with virtually no benefits

Reply:
If we change the type to `PropertiesField` within the TestData interfaces, it is going to force us to add in missing unnecessary properties. The casting has to stay. Maybe to add a bit more type safety closer to the source, all I can think of is to do some form of deep partial of `PropertiesField` and pick out the required properties or just leave this as is.

Reply:
Let's leave it as is. We don't want test data to be bloated too much.